### ISSUE TITLE
New condition: test for jurisdiction

### ISSUE THREAD

**Issue by georgd** [citation-style-language/schema#320]
**Is your feature request related to a problem? Please describe.**

Traditionally, each jurisdiction has its own tradition of formatting legislation and case law. Thus, for legal citation styles knowing the jurisdiction of the cited item is crucial. This is well covered in Juris-M with the MLZ CSL variant.

For non-juridic citation styles it seems that a single citation format for juridic material suffices but that’s not true in the European context. For members of the EU, EU legislation is an integral part of the national legislation. Therefore, anybody in the EU citing legal background should be able to apply the national and the EU formatting rules. E.g. a sociologist working on equality of opportunities in Austrian schools will need to cite the EU directive for equal treatment as well as Austrian school laws.

This doesn’t aim at adding lots of national implementations to every style but to facilitate the creation of styles that don’t require the full functionality of Juris-M styles but the ability to use two or three legal formats (e.g. in academic institutional styles).

**Describe the solution you'd like**
```xml
<choose>
  <if jurisdiction="at">
  ...
  </if>
  <else-if jurisdiction="eu.int">
  </else-if>
</choose>
```
This tests the contents of the jurisdiction variable but not allow arbitrary text. Rather it should be test for names according to the URN:LEX proposal. This is exactly the same as implemented in MLZ CSL and described here: <https://citeproc-js.readthedocs.io/en/latest/csl-m/index.html#jurisdiction-extension>
  

**Describe alternatives you've considered**
Currently, this is implemented in citation styles by testing for some unspecific field. This usually requires style specific input and is thus not desirable.

A solution would be to use Juris-M which however with its added complexity is often considered overkill.

**Comment by denismaier** [comment]
That's a good and valuable suggestion, yes. 
The problem is just that those tests will have to be added to each style right? Maybe we should consider adding this in a modularized form?

**Comment by georgd** [comment]
> The problem is just that those tests will have to be added to each style right?

No, I explicitly don’t see this as an aim at the moment. For globally used styles, this is impossible without modularising (and even that path is not really easy, think of EU legislation and localisation). I have styles in mind that target a restricted user group with a need for citing not more than a handful of different legislations (institutional styles, journals with a regional focus,...). So, I think, this can be introduced without adding the other. If a style author implements it for more legislations, it’s their choice – I wouldn’t and I don’t think the main CSL style contributors would do so and that’s reasonable.

**Comment by denismaier** [comment]
Ok. You know way more about legal citations than I. What would be needed for more complete legal citation support in CSL independent of CSLm? What are the core feature that we would have to adopt?

**Comment by georgd** [comment]
Now, that’s difficult to say. Considering you don’t want to replace CSLm for juridic styles but want to improve citing legal material in "general purpose" styles, I think adding this and the variables and types planned for 1.0.2, you’ve got a good set – for civil law. In these systems, nonlegal works usually cite laws and regulations (and similar) which are relatively simple and can be handled with what will exist in CSL according to the current state of accepted commits. Perhaps the legal `commenter` role might be useful in case a legal case be cited with its commenter in a journal.

I think nonlegal US (and other common law countries) texts tend to cite court decisions regularly. So, more of the solutions from CSLm might be necessary. But I don’t see myself expert enough to declare which. There are two main difficulties: courts and reporters on one side and parallel citations on the other.

**Comment by bwiernik** [comment]
We definitely don't want to add jurisdiction-specific legal formatting in vanilla CSL. That is firmly in the realm of CSLm--as you note, legal citation is much much more complex that academic citation, and supporting it requires much more complexity than is needed in non-legal styles. If such modularity is needed, uses the modular legal styles in CSLm is the way to go. 

I'm not sure what the `commenter` is. How are such citations handled in CSLm?

**Comment by georgd** [comment]
Again, this doesn’t aim at adding the whole bunch of complex features required for legal citation to CSL. That is well served in CSLm. This is about adding a functionality that is required for “simple” non-legal styles.

> We definitely don't want to add jurisdiction-specific legal formatting in vanilla CSL.

IMO it’s not a question _if_ you want to but _how_. Jurisdiction-specific legal formatting is a reality in non-legal vanilla CSL styles – it’s just disguised. The use case is detailed above (EU and national legislation support in a non-legal citation style, say for pedagogics). There are currently two ways one could implement that:

1. write a style definition that handles both cases without a conditional, that depends on users entering the same kind of metadata in different fields.
2. instruct the user to enter something in an arbitrary field and test that variable for input.

Both lead to style-specific data entry, both are not transparent to the user and are error prone. And for the style author who hasn’t thought to use code comments it might mean some hard time to find out why he did it this way.

> I'm not sure what the `commenter` is. How are such citations handled in CSLm?

The `commenter` is somebody who comments (contextualises, explains and critisizes) on a legal case or on an act of legislation either in a journal or in a special “commentary” which is an edited book structured like the law that is commented and every section of the law is commented by one or more commenters. CSLm has the name variable `commenter` for this.
